demos/python_demos/colorization_demo/colorization_demo.py

```diff
@@ -0,0 +1,145 @@
+
+    # mean is stored in the source caffe model and passed to IR
+    mean = get_parameters(config_xml)
+
+    log.basicConfig(format="[ %(levelname)s ] %(message)s",
+                    level=log.INFO if not args.verbose else log.DEBUG, stream=sys.stdout)
+
+    log.debug("Load network")
+    load_net = IENetwork(model=config_xml, weights=weights_bin)
+    load_net.batch_size = 1
+    exec_net = IECore().load_network(network=load_net, device_name=args.device)
+
+    assert len(load_net.inputs) == 1, "Number of inputs does not match network inputs"
+    input_blob = next(iter(load_net.inputs))
+    input_shape = load_net.inputs[input_blob].shape
+    assert input_shape[1] == 1, "Expected model output shape with 1 channel"
+
+    assert len(load_net.outputs) == 1, "Number of outputs does not match network outputs"
+    output_blob = next(iter(load_net.outputs))
+    output_shape = load_net.outputs[output_blob].shape
+    assert output_shape == [1, 313, 56, 56], "Shape of outputs does not match network shape outputs"
+
+    _, _, h_in, w_in = input_shape
+
+    try:
+        input_source = int(args.input)
+    except ValueError:
+        input_source = args.input
+
+    cap = cv.VideoCapture(input_source)
+    if not cap.isOpened():
+        assert "{} not exist".format(input_source)
+
+    color_coeff = np.load(coeffs).astype(np.float32)
+    assert color_coeff.shape == (313, 2), "Current shape of color coefficients does not match required shape"
+
+    while True:
+        log.debug("#############################")
+        hasFrame, original_frame = cap.read()
+        if not hasFrame:
+            break
+        (h_orig, w_orig) = original_frame.shape[:2]
+
+        log.debug("Preprocessing frame")
+        if original_frame.shape[2] > 1:
+            frame = cv.cvtColor(cv.cvtColor(original_frame, cv.COLOR_BGR2GRAY), cv.COLOR_GRAY2RGB)
+        else:
+            frame = cv.cvtColor(original_frame, cv.COLOR_GRAY2RGB)
+
+        img_rgb = frame.astype(np.float32) / 255.
+        img_lab = cv.cvtColor(img_rgb, cv.COLOR_RGB2Lab)
+        img_l = img_lab[:, :, 0]
+
+        img_rs = cv.resize(img_rgb, (w_in, h_in))
+        img_lab_rs = cv.cvtColor(img_rs, cv.COLOR_RGB2Lab)
+        img_l_rs = img_lab_rs[:, :, 0]
+        img_l_rs -= mean
+
+        log.debug("Network inference")
+        res = exec_net.infer(inputs={input_blob: [img_l_rs]})
+
+        (n_out, c_out, h_out, w_out) = res[output_blob].shape
+        update_res = (res[output_blob] * color_coeff.transpose()[:, :, np.newaxis, np.newaxis]).sum(1)
+
+        log.debug("Get results")
+        out = update_res.transpose((1, 2, 0))
+        out = cv.resize(out, (w_orig, h_orig))
+        img_lab_out = np.concatenate((img_l[:, :, np.newaxis], out), axis=2)
+        img_bgr_out = np.clip(cv.cvtColor(img_lab_out, cv.COLOR_Lab2BGR), 0, 1)
+
+        if not args.no_show:
+            log.debug("Show results")
+            imshowSize = (640, 480)
```

Comment:
Why not use the original image size?

Reply:
Image can be large. I change average size for show 3 image per monitor.